rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricFileFacts.java

```diff
@@ -114,6 +119,9 @@ private class FileFact {
         private volatile List<Diagnostic> typeCheckerMessages = Collections.emptyList();
         private final ParametricSummaryBridge summary;
 
+        private final ScheduledExecutorService calculateScheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
we normally share which executor we use, so that we can schedule clean shutdowns. Is there a reason we need a different pool here?

Reply:
Fixed as discussed (but added a helper class in `util.concurrent` to ensure that tasks scheduled using `ScheduledExecutorService` are guaranteed to be executed by the existing executor after the delay)